api/spec/requests/api/v1/entitlements_spec.rb

```diff
@@ -0,0 +1,63 @@
+require "swagger_helper"
+
+RSpec.describe "Entitlements", type: :request do
+  let(:project) { FactoryBot.create(:project) }
+  let(:project_id) { project.id }
+  let(:resource) { FactoryBot.create(:entitlement, :project_read_access, subject: project) }
+  let(:id) { resource.id }
+
+  let!(:target_user) { FactoryBot.create(:user) }
+  let(:target_url) { target_user.to_gid.to_s }
+  let(:subject_url) { project.to_entitlement_gid.to_s }
+  let(:scoped_roles) { { read_access: true } }
+
+  path "/entitlements" do
+    include_examples "an API create request",
+      model: Entitlement,
+      authorized_user: :admin do
+        let(:body) do
+          {
+            data: {
+              attributes: {
+                subject_url: subject_url,
+                target_url: target_url,
+                scoped_roles: scoped_roles
+              }
+            }
+          }
+        end
+      end
+
+    include_examples "an API index request", model: Entitlement
+  end
+
+  path "/entitlements/{id}" do
+    include_examples "an API destroy request", model: Entitlement, authorized_user: :admin
+    include_examples "an API show request", model: Entitlement, authorized_user: :admin
+  end
+
+  context "for a project" do
+    path "/projects/{project_id}/relationships/entitlements" do
+      include_examples "an API create request",
+        parent: "project",
+        model: Entitlement,
+        url_parameters: [:project_id],
+        authorized_user: :admin,
+        description: "Creates an entitlement to be associated with the project "\
+        "ID provided in the endpoint. The subjectUrl in this case is not needed." do
```

Comment:
The route will always accept a subjectUrl as an argument, but it has no function in this route, so I just documented that it is not needed here. Is that overkill? Should the subjectUrl have documentation in the resource information at the bottom of the page and then leave the user to understand when it is required and when it is unneeded data?

Reply:
I wouldn't document it as part of this route. Parameters that are accepted but not used don't need to be documented.